HTML/Default/xmlbrowser.html

```diff
@@ -170,7 +170,7 @@
 		[% IF item.favorites == 1 %]
 			[% WRAPPER favaddlink noTarget=1 %]
 				[% IF songinfo.favorites && item.type != 'audio' %]
-				onclick="SqueezeJS.Utils.toggleFavorite(this, '[% songinfo.favorites_url | uri | replace("'", "%27") %]', '[% crumb.name | html | replace("'", "%27") %]');"
+				onclick="SqueezeJS.Utils.toggleFavorite(this, '[% songinfo.favorites_url | uri | replace("'", "%27") %]', '[% songinfo.favorites_title | html | replace("'", "%27") %]', '[% songinfo.favorites_icon | uri | replace("'", "%27") %]');"
```

Comment:
This will only work if the underlying plugin does implement the `favorites_title`, right? Which basically means we break all of them, lack of backwards compatibility.

Reply:
No no! I'm not that careless, I'm might miss things, but I'm very aware of backward compatibility 😄

If a plugin sets favorites_icon, it will be used, otherwise it will use what was already in the feed. In fact, this fixes an issue we have now (even before all our PR and discussions) with adding a favorite using the heart icon when you are looking at track details through the "More" menu (remember, we discussed that this has been broken forever, this was one of the reason for @darrell-k initial PR, the fact that you have a random value being inserted in the favorite title. Because, the code was using crumb.values.-1 which was non-deterministic. All that is set in Web::XMLBrowser, when songinfo is set

Reply:
I'm sorry, I edited my comment after I saw the XMLBrowser change. What about my edited comment about other interfaces?

Reply:
Other interface don't use the songinfo key that is set in the Web::XMLBrowser, but I don't think it was your question. The favorites_title key set in there does not matter then. Now, in Control::XMLBrowser, one of my previous PR sets a favorites_title key in the JSON payload based on what it was before (see sub _favoritesParams), just using first the $item->{favorites_title} if any. 

So it was already bringing the logic if "if you set a favorites_titles in the feed's items, it will take precedence".

Reply:
And to more precisely answer your question, the "favorites_title" alwasy existed and the other UI have been using it (see https://github.com/LMS-Community/slimserver/blame/8b7b3bedd4f34ceeef4435a71377c14c30d5840e/Slim/Control/XMLBrowser.pm#L1917) but it was set to $item->title. All I added back then was the option to set as *in the item* a favorites_title key that would take precedence. Otherwise, in other UI, we could only display the item title (e.g. the track name but not the track *and* the artist)